R/peakdet.R

```diff
@@ -1,109 +1,112 @@
+#' 
+#' @param rtime `numeric` with retention times.
+#' 
+#' @param intensity `numeric` (same length as `rtime`) with the signal
+#'     intensities.
+#' 
+#' @param targetRtime `numeric(1)` with the expected retention time of the
+#'     target peak.
+#'
+#' @param .check `logical(1)` whether input parameters should be checked. Use
+#'     `.check = FALSE` only if validity of the input parameters was checked
+#'     in an upstream function. 
+#' 
+#' @author Pablo Vangeenderhuysen
+#' 
 #' @return
-#' @export
 #'
-
-find_peak_points <- function(rtvector, vector, searchrt) {
-  # Compute the derivative of the vector
-
-
-  # Find the indices where the derivative changes sign
-  sign_changes <-  c(FALSE, diff(diff(vector)>0)!=0)
-
-  # Find absolute maximum
-  peak_index <- which.max(vector)
-
-  # Find the local maxima
-
-  all_local_max <- which(diff(sign(diff(vector)))==-2)+1
-
-
-  # Delete local maxima with an intensity lower than 50k
-
-  local_max <- c()
-
-  for(lmax in all_local_max){
-    if(vector[lmax] > 0.5* max(vector[all_local_max])){
-      local_max <- cbind(local_max,lmax)
+#' @importFrom xcms imputeLinInterpol
+#' 
+#' @export
+find_peak_points <- function(rtime = numeric(), intensity = numeric(),
+                             targetRtime = numeric(), .check = TRUE) {
+    if (.check) {
+        if (length(rtime) != length(intensity))
+            stop("'rtime' and 'intensity' are expected to have the same length")
+        if (length(targetRtime) != 1L)
+            stop("'targetRtime' is expected to be a single numeric")
     }
-  }
-
-
-  # Find the max intensity closest to the searchrt which is at least 50% of the max intensity in the window
-
-  maxrts <- rtvector[c(peak_index,local_max)]
-
-
-  differences <- abs(maxrts - searchrt)
-
-  # Find the position (index) with the minimum difference
-  closest_position <- which.min(differences)
-
-  peak_index <- which(rtvector == maxrts[closest_position])
-
-  # Find the left and right border points from the peak
-  occur <- find_true_occurrence(sign_changes,vector,peak_index)
+    ## Compute the derivative of the vector jo: is here something missing?
+    if (anyNA(intensity))
+        intensity <- imputeLinInterpol(intensity)
+    di <- diff(intensity)
+    sign_changes <- c(FALSE, diff(di > 0) != 0, FALSE) # same length than ints
+    peak_index <- which.max(intensity)
+    all_local_max <- which(diff(sign(di)) == -2) + 1
+    ## Delete local maxima with an intensity lower than 50% max int
+    local_max <- all_local_max[intensity[all_local_max] >
+                               0.5 * intensity[peak_index]]
+    ## Find the local maximum closest to the targetRtime 
+    differences <- abs(rtime[local_max] - targetRtime)
+    peak_index <- local_max[which.min(differences)]
+    ## Find the left and right border points from the peak
+    border <- .find_peak_border(sign_changes, peak_index)
+
+    c(border, peak_index = peak_index)
```

Comment:
Changed the result from a `list` to a `integer`. Should be more efficient and consistent.